travis-ci/travis-cookbooks
# Turn off mysql performance schema

Please make sure you cover the following points:

### What is the problem that this PR is trying to fix?
We have had some reports of performance degradation between MySQL 5.5 (Precise) and 5.6 (Trusty) - see https://github.com/travis-pro/team-blue/issues/728.
One major difference between the two versions is the fact the performance_schema is turned on by default in 5.6, which causes some disk and memory overhead.

### What approach did you choose and why?
Since it's not particularly useful for test builds (and it wasn't available previously anyway), turn off the performance schema by default. (Instructions and more details can be found in the [mysql docs](https://dev.mysql.com/doc/refman/5.6/en/performance-schema-quick-start.html))
I initially wanted to add the config option to the existing [.my.cnf](https://github.com/travis-ci/travis-cookbooks/blob/master/cookbooks/travis_build_environment/templates/default/ci_user/dot_my.cnf.erb) template, but that didn't work, since this particular option requires root access to take effect.
Adding an extra template config.d did the trick.

Additionally, I changed the order of the steps in the recipe to create both configuration files before starting the service - otherwise a restart would be needed for the additional options to be applied.

### How can you test this?
Build a new image and check that the performance schema was turned off for mysql. 
You can check that by running the following query:
```
mysql> SHOW VARIABLES LIKE 'performance_schema';

+--------------------+-------+

| Variable_name      | Value |

+--------------------+-------+

| performance_schema | OFF   |

+--------------------+-------+
```

I managed to successfully build garnet using this test-branch: https://travis-ci.org/travis-infrastructure/packer-build/builds/266588053
Using that image, the memory footprint of mysql on GCE was reduced from 500MB to around 50MB.
